### PR TITLE
Jxp chapter7

### PR DESCRIPTION
**Pull request by JPeace2**

### CODE REVIEW

File: chapter07/jxp/ch7ex4.pl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+while ( my $line = <> ){
+# Just checks that at least one character is lower case. If you want to check there is just
+# one word and only the first letter can be a capital use ^[A-Z]?[a-z]+$ in egrep (but not this script).
+  if ( $line =~ /[a-z]/ ){
```
Comment (MarkWignall) [inline chapter07/jxp/ch7ex4.pl:8]: This seems it would pick up words without capitals? i.e tim. Why can your comment be used in egrep but not here?

Comment (JPeace2) [inline chapter07/jxp/ch7ex4.pl:8]: Yes, I misread the question so I just checked that the entire word wasn't capitalised. As for egrep that's a question for ard or ajd.

File: chapter07/jxp/ch7ex6.pl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+while ( my $line = <> ){
+  if ( $line =~ /wilma.+fred|fred.+wilma/ ){
```
Comment (codavex) [inline chapter07/jxp/ch7ex6.pl:6]: Nicely done, don't think anyone else did it this way

File: chapter07/jxp/ch7ex5.pl [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+while ( my $line = <> ){
+  if ( $line =~ /(\S)\g{1}/ ){
```
Comment (qianliok) [inline chapter07/jxp/ch7ex5.pl:6]: is `\g{1}` the same as `\1`?

File: chapter07/jxp/ch7ex1.pl [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+while ( my $line = <> ){
```
Comment (qianliok) [inline chapter07/jxp/ch7ex1.pl:5]: Nice, I like this while loop with `$line` assignment

**Pull request merged**